### ISSUE TITLE
Game hangs with SESSION CALLBACK EXCEPTION after 10 minutes

### ISSUE THREAD

**Issue by timhae** [magefree/mage#7142]
I have a fork of xmage which follows master, I usually deploy a new version of xmage client/server every week or so. In the last two weeks I noticed, that most games hang after about 10 minutes with the following message in the server log:
```
Oct 21 21:47:00 MYSERVER xmage_server[1869]: WARN  2020-10-21 21:47:00,206 SESSION CALLBACK EXCEPTION - MYUSER userId 807ca335-2056-419d-a5e7-75160c40a4d
e messageId: 372 - cause: java.io.IOException: Connection is closed =>[GAME 02f1a4a0-5608-4787-b702-15f2943bf470] Session.lambda$fireCallback$2
Oct 21 21:48:32 MYSERVER xmage_server[1869]: WARN  2020-10-21 21:48:32,587 Game frozen in waitResponseOpen for user MYUSER (connection problem)          
           =>[CALL main-169] HumanPlayer.waitResponseOpen
Oct 21 21:48:32 MYSERVER xmage_server[1869]: WARN  2020-10-21 21:48:32,745 Game frozen in waitResponseOpen for user MYUSER (connection problem)          
           =>[CALL main-238] HumanPlayer.waitResponseOpen
Oct 21 21:48:32 MYSERVER xmage_server[1869]: WARN  2020-10-21 21:48:32,897 Game frozen in waitResponseOpen for user MYUSER (connection problem)          
           =>[CALL main-239] HumanPlayer.waitResponseOpen
```
The game is frozen for all players, the GUI not though. I am reverting to the latest release right now, I will comment later whether the issue persists or not.

**Comment by JayDi85** [comment]
Check that user. If he used proxy then that's ok -- some proxy can break connections after some timeout (e.g. go to sleep until next request). It works fine with browser, but bad for app. XMage needs stable connection with server.

**Comment by JayDi85** [comment]
BTW, if you build own client-server from sources then you must change versions. Dev sources INCOMPATIBLE with official clients and sometime with old betas (that's why user freezed -- he got empty battlefield).

![shot_201022_090909](https://user-images.githubusercontent.com/8344157/96827305-59b17380-1446-11eb-975a-ff662f78a9c1.png)

**Comment by timhae** [comment]
Thanks for the feedback, I still have the same issue with the latest release tag so I am closing this since its only an issue for me. I do have a very unreliable connection since I am using my phone as wifi hotspot but unfortunately there is no other way at the moment. I am also hosting my own server so there is no version mismatch but I have noticed the empty battlefield whenever I updated the client and forgot about the server :P thanks for getting back so quick!
